hypothesis-python/tests/cover/test_float_nastiness.py

```diff
@@ -261,3 +261,24 @@ def test_can_exclude_neg_infinite_endpoint(x):
 @given(st.floats(1e307, float("inf"), exclude_max=True))
 def test_can_exclude_pos_infinite_endpoint(x):
     assert not math.isinf(x)
+
+
+def test_exclude_infinite_endpoint_is_invalid():
+    with pytest.raises(InvalidArgument):
+        st.floats(min_value=float("inf"), exclude_min=True).validate()
+    with pytest.raises(InvalidArgument):
+        st.floats(max_value=float("-inf"), exclude_max=True).validate()
+
+
+def test_exclude_entire_interval():
+    with pytest.raises(InvalidArgument, match="exclude_min=.+ and exclude_max="):
+        st.floats(0, 0, exclude_min=True, exclude_max=True).validate()
+
+
+@checks_deprecated_behaviour
+@given(data=st.data(), midpoint=st.floats(min_value=1, allow_infinity=False))
+def test_exclusions_tiny_interval(data, midpoint):
+    hi = Decimal(next_up(midpoint)) - Decimal("10e-309")
```

Comment:
What's the significance of `10e-309`? Worth a comment?

Reply:
That, and `min_value=1`, both related to minimum nonzero normal floats.  Now heavily commented and without the min_value, as we can just crank up the exponent and deal with subnormal floats too.